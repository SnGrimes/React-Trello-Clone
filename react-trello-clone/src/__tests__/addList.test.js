import React from 'react';
import { shallow } from 'enzyme'
import AddList from '../components/addList';

  describe('AddList', () => {
    const mockCreate = jest.fn();
    const mockCancel = jest.fn(); 
    const mockAdd = jest.fn();
    const mockCanAdd = jest.fn();
    const props = {cancelList: mockCancel, addList: mockAdd};
    const addList = shallow(<AddList {...props}/>);

    it('renders correctly', () => {
      expect(addList).toMatchSnapshot();
    });
    it('initializes the `value` with an empty string', () => {
      expect(addList.state().value).toEqual('');
    });
    describe('when clicking the `list_button-submit` button', () => {
      const form = addList.find('form');
      beforeEach(() => {
        addList.find('.list__button-submit').simulate('click');
      });
      it('calls the onsubmit event', () => {
        form.simulate('submit', { preventDefault() {} });
      });

    }); // list_button-submit describe block
    // describe('when clicking the `x` button', () =>{
    //   beforeEach(() => {
    //     addList.find('.list_x').simulate('click');
    //   });
    //   it('calls the cancelAdd function', () => {
    //     expect(mockCancel).toBeCalled();
    //   });
    //}); // x button describe block

  }); // main AddList describe block
 