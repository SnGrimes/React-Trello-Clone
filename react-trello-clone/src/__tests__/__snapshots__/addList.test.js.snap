// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddList
    addList={[Function]}
    cancelList={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <input
                name="listName"
                onChange={[Function]}
                placeholder="Add a list..."
                size="30"
                type="text"
                value=""
        />
        <button
                className="list__button-submit success button small"
                type="submit"
        >
                Save
        </button>
        <button
                className="list__x"
                onClick={[Function]}
        >
                <svg
                        className="feather feather-x"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                >
                        <line
                                x1="18"
                                x2="6"
                                y1="6"
                                y2="18"
                        />
                        <line
                                x1="6"
                                x2="18"
                                y1="6"
                                y2="18"
                        />
                </svg>
        </button>
</form>,
      "className": "list__add cell small-2",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="listName"
            onChange={[Function]}
            placeholder="Add a list..."
            size="30"
            type="text"
            value=""
/>,
          <button
            className="list__button-submit success button small"
            type="submit"
>
            Save
</button>,
          <button
            className="list__x"
            onClick={[Function]}
>
            <svg
                        className="feather feather-x"
                        fill="none"
                        height="24"
                        stroke="currentColor"
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth="2"
                        viewBox="0 0 24 24"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
            >
                        <line
                                    x1="18"
                                    x2="6"
                                    y1="6"
                                    y2="18"
                        />
                        <line
                                    x1="6"
                                    x2="18"
                                    y1="6"
                                    y2="18"
                        />
            </svg>
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "listName",
            "onChange": [Function],
            "placeholder": "Add a list...",
            "size": "30",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "className": "list__button-submit success button small",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="feather feather-x"
              fill="none"
              height="24"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
>
              <line
                            x1="18"
                            x2="6"
                            y1="6"
                            y2="18"
              />
              <line
                            x1="6"
                            x2="18"
                            y1="6"
                            y2="18"
              />
</svg>,
            "className": "list__x",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
/>,
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
/>,
              ],
              "className": "feather feather-x",
              "fill": "none",
              "height": "24",
              "stroke": "currentColor",
              "strokeLinecap": "round",
              "strokeLinejoin": "round",
              "strokeWidth": "2",
              "viewBox": "0 0 24 24",
              "width": "24",
              "xmlns": "http://www.w3.org/2000/svg",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "x1": "18",
                  "x2": "6",
                  "y1": "6",
                  "y2": "18",
                },
                "ref": null,
                "rendered": null,
                "type": "line",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "x1": "6",
                  "x2": "18",
                  "y1": "6",
                  "y2": "18",
                },
                "ref": null,
                "rendered": null,
                "type": "line",
              },
            ],
            "type": "svg",
          },
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <input
                    name="listName"
                    onChange={[Function]}
                    placeholder="Add a list..."
                    size="30"
                    type="text"
                    value=""
          />
          <button
                    className="list__button-submit success button small"
                    type="submit"
          >
                    Save
          </button>
          <button
                    className="list__x"
                    onClick={[Function]}
          >
                    <svg
                              className="feather feather-x"
                              fill="none"
                              height="24"
                              stroke="currentColor"
                              strokeLinecap="round"
                              strokeLinejoin="round"
                              strokeWidth="2"
                              viewBox="0 0 24 24"
                              width="24"
                              xmlns="http://www.w3.org/2000/svg"
                    >
                              <line
                                        x1="18"
                                        x2="6"
                                        y1="6"
                                        y2="18"
                              />
                              <line
                                        x1="6"
                                        x2="18"
                                        y1="6"
                                        y2="18"
                              />
                    </svg>
          </button>
</form>,
        "className": "list__add cell small-2",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="listName"
              onChange={[Function]}
              placeholder="Add a list..."
              size="30"
              type="text"
              value=""
/>,
            <button
              className="list__button-submit success button small"
              type="submit"
>
              Save
</button>,
            <button
              className="list__x"
              onClick={[Function]}
>
              <svg
                            className="feather feather-x"
                            fill="none"
                            height="24"
                            stroke="currentColor"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="2"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
              >
                            <line
                                          x1="18"
                                          x2="6"
                                          y1="6"
                                          y2="18"
                            />
                            <line
                                          x1="6"
                                          x2="18"
                                          y1="6"
                                          y2="18"
                            />
              </svg>
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "listName",
              "onChange": [Function],
              "placeholder": "Add a list...",
              "size": "30",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save",
              "className": "list__button-submit success button small",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Save",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <svg
                className="feather feather-x"
                fill="none"
                height="24"
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
>
                <line
                                x1="18"
                                x2="6"
                                y1="6"
                                y2="18"
                />
                <line
                                x1="6"
                                x2="18"
                                y1="6"
                                y2="18"
                />
</svg>,
              "className": "list__x",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <line
                    x1="18"
                    x2="6"
                    y1="6"
                    y2="18"
/>,
                  <line
                    x1="6"
                    x2="18"
                    y1="6"
                    y2="18"
/>,
                ],
                "className": "feather feather-x",
                "fill": "none",
                "height": "24",
                "stroke": "currentColor",
                "strokeLinecap": "round",
                "strokeLinejoin": "round",
                "strokeWidth": "2",
                "viewBox": "0 0 24 24",
                "width": "24",
                "xmlns": "http://www.w3.org/2000/svg",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "x1": "18",
                    "x2": "6",
                    "y1": "6",
                    "y2": "18",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "line",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "x1": "6",
                    "x2": "18",
                    "y1": "6",
                    "y2": "18",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "line",
                },
              ],
              "type": "svg",
            },
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
